File: Aula 3/Exercício 1.js
// class Usuario {
//     constructor(nome, email) {
//         this.nome = nome;
//         this.email - email;
//     }
// }

// ================================================================================
// Exercício 1 - Classe Produto (atribútos e método)
// ================================================================================

class Produto {
    constructor(nome, preco, estoque) {
        this.nome = nome;
        this.preco = preco;
        this.estoque = estoque;
    }

    exibirInfo() {
        console.log(`${this.nome}: R$${this.preco} | ${this.estoque}.`)
    }
}

// Instanciando e chamando
const p = new Produto("Café", 18.90, 50);
p.exibirInfo();

// ================================================================================
// Exercício 2 - Interação entre objetos (Pedido recebe Produto)
// ================================================================================

class Pedido {
    constructor() {
        this.itens = [];
    }

    adicionarProduto(produto) {
        this.itens.push(produto.nome);
        consol.log(`Adicionado: ${produto.nome}`);
    }
}

// Criando os dois objetos
const cafe = new Produto("Café", 18.90, 50);
const pedido = new Pedido();

//Interação: pedido recebe café

File: Aula 3/Exercício 1.py
class Produto:
    def __init__(self, nome, preco, estoque):
        self.nome = nome
        self.preco = preco
        self.estoque = estoque
    
    def exibir_info(self):
        print(f"{self.nome}: R${self.preco} | {self.estoque} un.")



class Pedido:
    def __init__(self):
        self.itens = []

    def adicionar_produto(self, produto:Produto):
        self.itens.append(produto.nome)
        print(f"Adcionado: {produto.nome}.")

cafe = Produto("Café", 18.9, 50)
pedido = Pedido()

cafe.exibir_info()
pedido.adicionar_produto(cafe)